### ISSUE TITLE
Clarification on moco.py

### ISSUE THREAD

**Issue by Valensicv** [wvangansbeke/Unsupervised-Classification#44]
Hi, just wanted to highlight a possible, issue. In moco.py at line 50
`train_dataloader = get_val_dataloader(p, train_dataset)`
shoundn't the `get_train_dataloader` be used instead of `get_val_dataloader`?

Thank you.

**Comment by wvangansbeke** [comment]
Hi @Valensicv,

Thank you for trying out our code.

This might seem confusing but is actually correct. We are looking for the nearest neighbors on the train set (i.e. `train_dataset`) but use the validation dataloader. The validation dataloader won't drop the last batch as we want to find the nearest neighbors for all the samples in the training set. More importantly, we want to keep the original order when indexing the neighbors. In this way, the memory bank won't be shuffled.

Hope this helps.

**Comment by Valensicv** [comment]
Thank @wvangansbeke  for the prompt reply, and thanks for sharing this project. Now I see, and it is clear. thank you.
